2-Pricing/src/components/CheckoutRow.tsx

```diff
@@ -0,0 +1,53 @@
+import { Cart } from "../App";
+import { Subscription } from "../constants";
+import { SubscriptionKey } from "./Cart";
+
+interface CheckoutRowProps {
+  isMonthlyView: boolean;
+  subscription: Subscription;
+  inCart: Cart;
+  setInCart: (cart: Cart) => void;
+}
+
+export const CheckoutRow = ({
+  isMonthlyView,
+  subscription,
+  inCart,
+  setInCart,
+}: CheckoutRowProps) => {
+  const key = subscription.name.toLowerCase() as SubscriptionKey;
+  const quantityInCart = inCart[key] ?? 0;
+
+  const handleIncrementCart = () => {
+    const newCart = { ...inCart };
+    newCart[key] = quantityInCart + 1;
+    setInCart(newCart);
+  };
+
+  const handleDecrementCart = () => {
+    const newCart = { ...inCart };
+    if (quantityInCart < 2) delete newCart[key];
+    else newCart[key] = quantityInCart - 1;
+    setInCart(newCart);
```

Comment:
Same here

Reply:
I think it would make more sense to pass `handleDecrementCart` and `handleIncrementCart` as parameters to `CheckoutRow` instead of `inCart` and `setInCart`.

By doing this, we avoid coupling this component to the cart state.